byterun/minor_gc.c

```diff
@@ -488,20 +494,18 @@ void caml_empty_minor_heap_domain_finalizers (struct domain* domain, void* unuse
   struct caml_minor_tables *minor_tables = domain_state->minor_tables;
   struct caml_custom_elt *elt;
 
-  caml_ev_begin("minor_gc/finalisers");
   caml_final_update_last_minor(domain);
   /* Run custom block finalisation of dead minor values */
   for (elt = minor_tables->custom.base; elt < minor_tables->custom.ptr; elt++) {
     value v = elt->block;
-    if (Hd_val(v) == 0) {
+    if (get_header_val(v) == 0) {
```

Comment:
Do we need to update assert statements in the code that use `Hd_val(v)` to also use `get_header_val(v)` (e.g. in `next_minor_block`)?
Or even better, I think `next_minor_block` is now dead code and can be removed.

Reply:
You're right - I've removed `next_minor_block` and I think I've cleaned up any of the asserts I can find that use `Hd_val`.